include/defaultinitalloc.hpp

```diff
@@ -0,0 +1,38 @@
+/**
+ * Allocator adaptor that interposes construct() calls to convert value-initialization
```

Comment:
Once this is stable I'd like to benchmark it and see if omitting this custom allocator makes a visible difference.

Reply:
Lump this with #653, then.

Reply:
Sure, since you already wrote the file, but it's a chunk of complex C++ I'd rather we not have added at all unless/until after profiling suggested it would be worth doing.

Reply:
Unresolving until this is sorted out.